apiserver/pkg/util/cluster.go

```diff
@@ -216,17 +216,6 @@ func buildWorkerPodTemplate(imageVersion string, envs map[string]string, spec *a
 			Annotations: buildNodeGroupAnnotations(computeRuntime, spec.Image),
 		},
 		Spec: v1.PodSpec{
-			InitContainers: []v1.Container{
```

Comment:
sorry for late. BTW, try to get more background on this. The workers will result in failure at the begining, even it will restart, it will become misleading from observability perspective.  Is that a concern?

Reply:
Currently, the old init container logic is wrong. It waits for the head service rather than GCS server. The head service will be ready when the image pull finishes. It will not fail in regular conditions because Ray internally will retry multiple times to connect to the GCS server. However, the retry mechanism in Ray has a timeout, so the workers will fail if the GCS server cannot be ready within the timeout.

In #973, we inject a default init container using `ray health-check` to wait until the GCS server is ready. We removed the `nslookup` init container after release v0.5.0 to keep the compatibility philosophy of KubeRay (See #940 for more details).

Reply:
Hi, I also have a qq about this change, it seems like it prevents helm chart users to configre docker image to be used for the init container. I there something I'm missing? We have a strong requirement to use internal docker registry so this makes the chart 0.5.0 not usable as-is for us. I checked that the init container is still created and not removed completely

Reply:
Hi @skliarpawlo, your question is unrelated to KubeRay APIServer, so I open a new thread to answer your question.